src/libraries/System.Private.CoreLib/src/System/Int128.cs

```diff
@@ -1273,76 +1241,38 @@ public static Int128 Log2(Int128 value)
         /// <inheritdoc cref="IMultiplyOperators{TSelf, TOther, TResult}.op_Multiply(TSelf, TOther)" />
         public static Int128 operator *(Int128 left, Int128 right)
         {
-            // We simplify the logic here by just doing unsigned multiplication on
-            // the one's complement representation and then taking the correct sign.
-
-            ulong sign = (left._upper ^ right._upper) & (1UL << 63);
-
-            if (IsNegative(left))
-            {
-                left = ~left + 1U;
-            }
-
-            if (IsNegative(right))
-            {
-                right = ~right + 1U;
-            }
-
-            UInt128 result = (UInt128)(left) * (UInt128)(right);
-
-            if (result == 0U)
-            {
-                sign = 0;
-            }
-            else if (sign != 0)
-            {
-                result = ~result + 1U;
-            }
-
-            return new Int128(
-                result.Upper | sign,
-                result.Lower
-            );
+            // Multiplication is the same for signed and unsigned provided the "upper" bits aren't needed
+            return (Int128)((UInt128)(left) * (UInt128)(right));
         }
 
         /// <inheritdoc cref="IMultiplyOperators{TSelf, TOther, TResult}.op_CheckedMultiply(TSelf, TOther)" />
         public static Int128 operator checked *(Int128 left, Int128 right)
         {
-            // We simplify the logic here by just doing unsigned multiplication on
-            // the one's complement representation and then taking the correct sign.
-
-            ulong sign = (left._upper ^ right._upper) & (1UL << 63);
-
-            if (IsNegative(left))
-            {
-                left = ~left + 1U;
-            }
+            Int128 upper = BigMul(left, right, out Int128 lower);
 
-            if (IsNegative(right))
+            if (((upper != 0) || (lower < 0)) && ((~upper != 0) || (lower >= 0)))
             {
-                right = ~right + 1U;
-            }
+                // The upper bits can safely be either Zero or AllBitsSet
+                // where the former represents a positive value and the
+                // latter a negative value.
+                //
+                // However, when the upper bits are Zero, we also need to
+                // confirm the lower bits are positive, otherwise we have
+                // a positive value greater than MaxValue and should throw
```

Comment:
This comment should be updated to match the user-suggested fix

Reply:
Fixed.